edge-modules/PaidModuleSample/src/Program.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace PaidModuleSample
+{
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Runtime.InteropServices;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Console.WriteLine("PaidModuleSample Main() started.");
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
+
+            IConfiguration configuration = new ConfigurationBuilder()
+               .SetBasePath(Directory.GetCurrentDirectory())
+               .AddJsonFile("config/appsettings.json", optional: true)
+               .AddEnvironmentVariables()
+               .Build();
+
+            string iotHubHostName = configuration.GetValue<string>("IOTEDGE_IOTHUBHOSTNAME");
```

Comment:
I would suggest we re-factor this code a bit. 
In Program.cs, we do the following - 
PurchaseInfo purchaseInfo = await PurchaseInfoProvider.Get();
// Validate purchaseInfo and print relevant info - 
// Purchase info found, success OR No purchase info found, etc. 
Maybe loop and call the same method a few times. 
Then we can hide all the details to get purchase info behind the PurchaseInfoProvider, which can be in a separate folder.